src/py42/services/_connection.py

```diff
@@ -195,6 +195,15 @@ def request(
                 # setting this manually speeds up read times
                 response.encoding = u"utf-8"
 
+            if response is not None:
+                debug.logger.debug(
+                    u"Response: status={}, data={}".format(
+                        response.status_code, response.text
+                    )
+                )
```

Comment:
The part from lines 207-208 should be moved here because we don't need to check if the `response is not None` twice.

```
            if response is not None:
                debug.logger.debug(
                    u"Response: status={}, data={}".format(
                        response.status_code, response.text
                    )
                )
                if 200 <= response.status_code <= 399:
                  return Py42Response(response)
           else:
              ...
```

Reply:
refactored. Similar check was done even in next few lines as well, encompassed all of them into a single block.